lib/timers.js

```diff
@@ -580,13 +580,15 @@ function processImmediate() {
   while (immediate) {
     domain = immediate.domain;
 
-    if (!immediate._onImmediate)
-      continue;
```

Comment:
Yep, it seems moving those lines above the if branch has no effect, we still get stuck.